service/src/test/java/org/databiosphere/workspacedataservice/dataimport/DefaultImportValidatorTest.java

```diff
@@ -128,29 +129,57 @@ void rejectsImportsFromOtherSources() {
   }
 
   @Test
-  void acceptsGsUrlsForRawlsJsonImports() {
+  void acceptsGsUrlsWithMatchingWorkspaceIdForRawlsJsonImports() {
     // Arrange
-    URI importUri = URI.create("gs://test-bucket/file");
+    URI importUri =
+        URI.create("gs://test-bucket/to-cwds/%s/random.json".formatted(destinationWorkspaceId));
     ImportRequestServerModel importRequest =
         new ImportRequestServerModel(TypeEnum.RAWLSJSON, importUri);
 
     // Act/Assert
     assertDoesNotThrow(() -> importValidator.validateImport(importRequest, destinationWorkspaceId));
   }
 
+  @Test
+  void rejectsGsUrlsWithCorrectBucketAndWorkspaceButUnexpectedFormat() {
+    // Arrange
+    URI importUri =
+        URI.create("gs://test-bucket/random/%s/random.json".formatted(destinationWorkspaceId));
+    ImportRequestServerModel importRequest =
+        new ImportRequestServerModel(TypeEnum.RAWLSJSON, importUri);
+
+    // Act/Assert
+    assertThatExceptionOfType(ValidationException.class)
+        .isThrownBy(() -> importValidator.validateImport(importRequest, destinationWorkspaceId))
+        .withMessageContaining("/to-cwds/%s/".formatted(destinationWorkspaceId));
+  }
+
   @Test
   void rejectGsUrlsWithMismatchingBucketForJsonImports() {
     // Arrange
-    URI importUri = URI.create("gs://rando-bucket/file");
+    URI importUri =
+        URI.create("gs://rando-bucket/to-cwds/%s/random.json".formatted(destinationWorkspaceId));
     ImportRequestServerModel importRequest =
         new ImportRequestServerModel(TypeEnum.RAWLSJSON, importUri);
 
     // Act/Assert
-    ValidationException err =
-        assertThrows(
-            ValidationException.class,
-            () -> importValidator.validateImport(importRequest, destinationWorkspaceId));
-    assertEquals("Files may not be imported from rando-bucket.", err.getMessage());
+    assertThatExceptionOfType(ValidationException.class)
+        .isThrownBy(
+            () -> importValidator.validateImport(importRequest, WorkspaceId.of(UUID.randomUUID())))
+        .withMessageContaining("Files may not be imported from rando-bucket.");
+  }
+
+  @Test
+  void rejectsGsUrlsWithMismatchingWorkspaceId() {
+    // Arrange
+    URI importUri = URI.create("gs://test-bucket/to-cwds/does-not-match-workspace-id/random.json");
```

Comment:
To clarify the intent of the test and also check that validation doesn't fail only because the workspace ID segmetn isn't a UUID, maybe:

```suggestion
    UUID otherWorkspaceId = UUID.randomUUID();
    URI importUri = URI.create("gs://test-bucket/to-cwds/%s/other-workspace-import.json".formatted(otherWorkspaceId));
```

Reply:
Cool, applied fix!